test/sqlalchemy/test_across_schema.py

```diff
@@ -1,29 +1,48 @@
 from sqlalchemy import Table, MetaData, testing
 from sqlalchemy.testing import fixtures
-
+from sqlalchemy.orm import sessionmaker, load_only
+from sqlalchemy import func
+from sqlalchemy import distinct
 
 class AcrossSchemaTest(fixtures.TestBase):
-    TEST_DATABASE = 'test_sqlalchemy_across_schema'
```

Comment:
As we discussed, I'm fine with some backwards-incompatibility here (since it's already kind of broken). But we should have some way to make cross-database references going forward (it may not be common, but remember that this capability did originate with a user request in #30). I think the way to do this in SQLAlchemy is to use a dotted schema name that includes the database: `schema=self.TEST_DATABASE + '.public'`, which means that we should recognize this and adjust our information_schema queries accordingly. But the base postgres dialect doesn't do this, so I'm not certain about this (maybe cross-database references are rare in postgres because people use schemas for this purposes instead. Our lack of user-created schemas makes cross-database support more important).

Reply:
Yes as you found out this does not work because the base dialect issues more queries on `pg_catalog` tables with the schema string as constrain on various pg_catalog columns.

I would like to move this PR forward nevertheless: the feature was broken in 2.0 *anyway* and this PR here is about ensuring that the code at least runs with 2.0 and 2.1.

I would really prefer to solve this problem by addressing https://github.com/cockroachdb/cockroach/issues/26443 

If we don't go in that direction we'll have no choice but to implement the entire sqlalchemy dialect from scratch.